src/lib/builtins/string.rs

```diff
@@ -766,6 +798,7 @@ pub fn create_constructor(global: &Value) -> Value {
     proto.set_field_slice("substr", to_value(substr as NativeFunctionData));
     proto.set_field_slice("valueOf", to_value(value_of as NativeFunctionData));
     proto.set_field_slice("matchAll", to_value(match_all as NativeFunctionData));
+    proto.set_field_slice("replace", to_value(replace as NativeFunctionData));
```

Comment:
You need to set the [`length`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/length) property of the function as well.

Reply:
I don't know if I understood that. Do you mean `str.replace.length` should return `2`? If so, would you say that `str.charAt.length` should return `1` then?

Reply:
Yes, to both your questions. A `function`'s length represents the number of "formal" parameters, e.g. those with no default value.

Reply:
Do the others functions in `String` already support that in boa or are you suggesting that they should? I'm locally getting `str.charAt.length == undefined`, but I may have an outdated branch...

Reply:
Unfortunately, some functions in boa don't have their length set because it is easily missable.

I have suggested an approach in #193 and would very much like we have something like that, so we don't miss like things like that.

Reply:
Nice. I'm working on other stuff here, but I'll make sure to check where that PR of yours goes.